src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Corax;
+using Raven.Client;
+using Raven.Server.Documents.Indexes.Persistence.Corax.WriterScopes;
+using Raven.Server.Json;
+using Raven.Server.Utils;
+using Sparrow.Extensions;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server;
+using Sparrow.Threading;
+using Voron;
+using Encoding = System.Text.Encoding;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Corax
+{
+    public sealed class CoraxDocumentConverter : ConverterBase
+    {
+        private readonly ByteStringContext _allocator;
+        private readonly Dictionary<Slice, int> _knownFields;
+        private static readonly byte[] _trueLiteral = Encoding.UTF8.GetBytes("true"); 
+        private static readonly byte[] _falseLiteral = Encoding.UTF8.GetBytes("false"); 
+
+        public CoraxDocumentConverter(
```

Comment:
what is the life time of this object?

Reply:
Persistence (initialization in CoraxPersistance).